rs/messaging/src/routing/stream_builder/tests.rs

```diff
@@ -90,20 +90,16 @@ fn reject_local_request() {
 
         // With a reservation on an input queue.
         let payment = Cycles::new(100);
-        let msg = generate_message_for_test(
+        let mut msg = generate_message_for_test(
             sender,
             receiver,
             CallbackId::from(1),
             "method".to_string(),
             payment,
         );
-        register_callback(
-            &mut canister_state,
-            msg.sender,
-            msg.receiver,
-            msg.sender_reply_callback,
-            msg.deadline,
-        );
+        let callback_id =
+            register_callback(&mut canister_state, msg.sender, msg.receiver, msg.deadline);
+        msg.sender_reply_callback = callback_id;
```

Comment:
Would it be possible to have `register_callback` directly take a mutable reference to the message and set the callback ID appropriately inside `register_callback`?

Reply:
Yes but I think it makes more sense the register a callback, get a `CallbackId` for it and then use that to generate the message rather than generating a message that uses some dummy value for the callback ID and then replace it when registering.

Reply:
If this also works without too much effort in terms of refactoring even better.